cloudprovider/cloudprovider.go

```diff
@@ -100,7 +100,11 @@ func GetInstance(cloud Interface, IP string) (instance *types.Instance, err erro
 	runningMutex.Lock()
 	running[IP] = time.Now()
 	runningMutex.Unlock()
-	defer func() { delete(running, IP) }()
+	defer func() {
```

Comment:
Yes I have that as well on my local, and deleting the lock for read as reads are safe usually. Haven't pushed it yet because I was trying to get my head around this.